### ISSUE TITLE
Canceling the option to change the password from your account trigger the update password pop-up

### ISSUE THREAD

**Issue by sv-sdeiac** [mozilla/fxa#876]
Environment: 
Platform: Windows 10, Android;
Build: stage server [Train 135], production;

Prerequisites:
Make sure you are logged with a valid account.

Steps to reproduce:
1. Go to https://accounts.firefox.com/settings;
2. Tap on 'Change' from the password line;
3. Write your password and the new one;
4. Tap on 'Cancel' button;

Expected results:
The Section is closed.

Actual results:
The section is closed and the pop-up option to update the password is presented.

Note:
- [video](https://drive.google.com/file/d/17prFxwxzOlD70Ax0-r--pWQbfN8ceUNO/view?usp=sharing)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-814)
┆Issue Number: FXA-814

**Comment by sv-sdeiac** [comment]
This issue is reproducible also if you tap the cancel button after typing a password on the account recovery->generate

**Comment by sv-sdeiac** [comment]
This issue is still reproducible using Train [142] and production.
Environment:
- Windows 10 x64;
- Nokia 6 (Android 7.1.1);
- Xiaomi Mi 8 Lite (Android 9).

**Comment by jodyheavener** [comment]
Hi! I have looked into this a bit. For me both the Firefox and 1Password extension password dialogues were triggering.

TLDR: I don't know what the best solution is, but I think I know why it's happening.

I was able to get the 1Password dialogue to stop showing up by adding `type="button"` to the Cancel button in the Password section.

The Firefox dialogue is a little trickier to solve, though, as we have dual actions occurring when the user clicks Cancel. But first, the issue in this case (from what I can tell) is that we are navigating forward with a changed password-type input value, so the browser is inferring that because we are navigating we might want to save that changed password value.

The fix here is to reset the value of the password field before navigating. The code _does_ try to do this, but there are two components in play that don't know about each other:

- The Cancel button is a [React component](https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/scripts/views/settings/change_password.jsx#L229-L232) that [resets the field values](https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/scripts/views/settings/change_password.jsx#L106-L118) when you click it.
- Because the Password section has its own URL (`/settings/change_password`), we additionally use Backbone to bind any [`.cancel` element click](https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/scripts/views/mixins/settings-panel-mixin.js#L17) to [take us back](https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/scripts/views/mixins/settings-panel-mixin.js#L55) to `/settings` via `history.pushState`.

If the two reliably happened one after the other, I don't think this would be problem.

This can be demonstrated in two ways:

1. A quick example: go ahead and type anything into the Old password field, then in your console execute `history.pushState({}, '', '/settings')`. The Firefox dialogue should show up because you changed the password value, and you navigated forward.
2. If you have this code base running locally, go ahead and wrap `this.navigate('settings')` in a `setTimeout` so it gets pushed to the bottom of the event loop. The Firefox dialogue should not show up even after entering a new password value.

GLHF.

**Comment by byrw** [comment]
Thanks to @jodyheavener idea, I've created a solution for this issue. Can I create a PR for it?

**Comment by byrw** [comment]
Here is the solution I used.

1-  Add `type=button` to the cancel button

2- In settings-panel-mixin.js, change the order of` this.clearInput()` to before `this.navigate('settings)`.

3- In settings-panel-mixin.js, change the function `clearInput()`, to make sure is actually working.
Before:
     ` formEl.reset();`
After:
```
for (let i = 0; i < formEl.length; i++) {
 formEl[i].value = '';
}
```
